hail/python/hailtop/utils/gcs_requester_pays.py

```diff
@@ -0,0 +1,34 @@
+from typing import Any, DefaultDict, Dict, FrozenSet, Optional, Tuple, Type, TypeVar, Union
+
+from hailtop.aiocloud.aiogoogle import GCSRequesterPaysConfiguration
+
+
+FS = TypeVar("FS")
+MaybeGCSRequesterPaysConfiguration = Optional[GCSRequesterPaysConfiguration]
+GCSRequesterPaysKey = Optional[Union[str, Tuple[str, FrozenSet[str]]]]
+
+
+class GCSRequesterPaysFSCache(DefaultDict[MaybeGCSRequesterPaysConfiguration, FS]):
+    def __init__(self, *args: Any, fs_constructor: Type[FS], default_kwargs: Dict[str, Any] = {}, **kwargs: Any) -> None:
```

Comment:
Is the only use the `*args` here the default factory in the `gcs_requester_pays_fs_cache` function? If so can we restrict to just that argument and add more as necessary down the line? Makes for nicer type checking.

Reply:
On a similar note, I don't see `**kwargs` used. Can we remove it until we need it?

Reply:
removed!